boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,725 @@
+    RetryType_ReservedBit_0    = (1 << 4), //!< Unused Bit 0
+    RetryType_ReservedBit_1    = (1 << 5), //!< Unused Bit 0
+    RetryType_ReservedBit_2    = (1 << 6), //!< Unused Bit 0
+    RetryType_ReservedBit_3    = (1 << 7), //!< Unused Bit 0
+    RetryType_Add_Offset_Current =
+        (1 << 8) //!< Random Offset Current Added to Sensed Current
+} RetryType_e;
+
+//! Enumeration for the global configurations types
+typedef enum
+{
+    GlobalConfigType_OvervoltageProtection_Disable =
+        (1 << 0), //!< Over-voltage Protection Disable
+    GlobalConfigType_CSNS0_Enable =
+        (1 << 1), //!< Current Sensing on Channel 0 Enable
+    GlobalConfigType_CSNS1_Enable =
+        (1 << 2), //!< Current Sensing on Channel 1 Enable
+    GlobalConfigType_Temperature_Sense_Enable =
+        ((1 << 1) | (1 << 2)), //!< Temperature Sensing Enable
+    GlobalConfigType_VDD_Fail_Enable =
+        (1 << 3), //!< VDD Failure Detection Enable
+    GlobalConfigType_Watchdog_Disable = (1 << 4), //!< SPI Watchdog Disable
+    GlobalConfigType_TrackHold_Sense_Enable =
+        (1 << 5), //!< Track and Hold Current Sensing Enable
+    GlobalConfigType_Parallel_Mode_Enable =
+        (1 << 6), //!< Parallel Channels Mode Enable
+    GlobalConfigType_PWM_Channel0_Enable =
+        (1 << 7), //!< Channel 0 PWM Function Enable
+    GlobalConfigType_PWM_Channel1_Enable =
+        (1 << 8) //!< Channel 1 PWM Function Enable
+} GlobalConfigType_e;
+
+//! Enumeration for the diagnostic configurations types
+typedef enum
+{
+    DiagnosticType_Cal0_Failure =
+        (1 << 0), //!< Channel 0 Internal Clock Calibration Failure
+    DiagnosticType_Cal1_Failure =
+        (1 << 1), //!< Channel 1 Internal Clock Calibration Failure
+    DiagnosticType_ExtClock_Failure = (1 << 2), //!< External Clock Failure
+    DiagnosticType_Input0_On        = (1 << 3), //!< Input 0 Pin On
+    DiagnosticType_Input1_On        = (1 << 4), //!< Input 1 Pin On
+    DiagnosticType_ID0              = (1 << 5), //!< Product ID Number Bit 0
+    DiagnosticType_ID1              = (1 << 6), //!< Product ID Number Bit 1
+    DiagnosticType_CONF0            = (1 << 7), //!< Channel 0 Config DC Motor
+    DiagnosticType_CONF1            = (1 << 8)  //!< Channel 1 Config DC Motor
+} DiagnosticType_e;
+
+struct Efuse
+{
+    bool         watch_dog_state;
+    StatusType_e status;
+    struct ChannelFaults
+    {
+        FaultType_e channel_0_faults;
+        FaultType_e channel_1_faults;
+    } faults;
+    struct EfuseChipSelect
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } chip_select;
+    struct EfuseCurrentSenseSync
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } current_sense_sync;
+};
+
+/**
+ * Initializes the SPI handle for the Efuses
+ * @param hspi Handle to the SPI peripheral used for the Efuses
+ */
+void Io_Efuse_Init(SPI_HandleTypeDef *const hspi);
```

Comment:
```suggestion
void Io_Efuse_Init(SPI_HandleTypeDef * hspi);
```
This doesn't do anything in header